### ISSUE TITLE
Subscription issues when trying to subscribe via email when already subscribed via wpcom account. 

### ISSUE THREAD

**Issue by automattic-ian** [Automattic/jetpack#17055]
#### Steps to reproduce the issue

1. Subscribe to a site via wpcom account. 
2. Attempt to subscribe to the same site with the email address associated with the wpcom account while not logged in or logged in. 
3. See error `Try again with a valid email address`

#### What I expected
I expected either a nudge to log in to WordPress.com or  a message that that email address was already following a site. 

#### What happened instead

Basically, if you're signed out of a WordPress.com account but try to subscribe with the email address for that account, you get an error Try again with a valid email address instead of a nudge to sign in to WordPress.com, or a valid follow action.
**Screenshots**

**Comment by jeherve** [comment]
Closing as duplicate of #5407. We really need to make some improvements to this flow.
